rancher/rancher
# Update Rancher Machine CLI invocations

## Issue: https://github.com/rancher/rancher/issues/40608

 
## Problem
The original problem this PR (in combination with https://github.com/rancher/machine/pull/224) addresses is the following
1. Create a new node-provisioned cluster (RKE1 or 2, doesn't matter)
2. After the cluster has been created, rotate the associated cloud credential (i.e. change it in the cloud provider and update it in the Rancher UI)
3. Try to delete the cluster. You'll notice that the nodes cannot be deleted even though Rancher has the most up-to-date cloud credential.

I solved this problem in Rancher machine by simply having Rancher machine look for and use the updated cloud credential if the `--update-config` flag is passed. Rancher Machine will still behave as it always has if that flag is not passed.
 
## Solution
This PR just updates Rancher to pass the `--update-config` flag on the `rancher-machine rm` command so Rancher Machine knows to re-load configuration from envvars (for RKE2) or CLI flags (for RKE1).
 
## Testing


## Engineering Testing
### Manual Testing
- Create RKE1 and 2 node-provisioned clusters
- Rotate credentials
- Try delete the clusters and ensure that all nodes are properly deleted

### Automated Testing

* Test types added/modified:
    * Unit: Included in https://github.com/rancher/machine/pull/224/files#diff-835f2809e3a2abc47591f343924d8981dc259d95f1f5eddeef8c7543cf5368ffR39
    * Integration: https://github.com/rancher/machine/pull/224/files#diff-835f2809e3a2abc47591f343924d8981dc259d95f1f5eddeef8c7543cf5368ffR39
  
Summary: I wrote unit and full CLI tests in my Rancher Machine PR to ensure that the CLI still works as intended.

## QA Testing Considerations

 
### Regressions Considerations

I think the main thing to focus on is the case where credentials are changed after cluster creation. This is relevant for all node drivers, so we probably want to test this scenario with each one to ensure it works as intended.

Existing / newly added automated tests that provide evidence there are no regressions:
